### PR TITLE
implement some dialogs

### PR DESCRIPTION
**Pull request by balcy**
java script dialogs (alert, unload, confirm...), file upload, geolocation, color (new), ...
note: media access (camera, mic) did not yet work (regardless of the dialog implementation)
not yet done: downloads handling / fullscreen request

### CODE REVIEW

**Comment by balcy** [comment]
note: the text select/copy/paste function not as good as in the old browser (dialogs should appear next to the text fields), so just interim version here until proper text selection works

File: src/app/WebViewImpl.qml [hunk 1]
```diff
@@ -213,6 +213,15 @@ WebEngineView {
                 }
                 else
                 {
+                    contextMenuStartScroll.x = webview.scrollPosition.x;
+                    contextMenuStartScroll.y = webview.scrollPosition.y;
+
+                    webview.runJavaScript("var elemContextMenu = document.elementFromPoint(%1, %2);
```
Comment (Flohack74) [inline src/app/WebViewImpl.qml]: Missing a " at the end?

**Review by Flohack74 — CHANGES_REQUESTED** [review]

**Comment by balcy** [comment]
should be fine, it is a two line java script command (the " is before the .arg() in the next line)
[edit] no idea why github makes those lines red, Qt Creator did not complain about them, and it worked ... to make it more readable, I've put the command to the variable "javaScriptCommand" now.

**Comment by balcy** [comment]
just do not trust github, but me (and QtCreator) here : )
even Qt tutorials get those red lines:
https://github.com/qt-labs/qml-presentation-system/blob/master/examples/tutorial/SlideDeck.qml

**Review by Flohack74 — APPROVED** [review]

**Comment by Flohack74** [comment]
Hmm you want to keep the PR open? You are still adding stuff...

**Comment by balcy** [comment]
For new stuff I can make a new PR
e.g. downloads are still missing.
But I will be on vacation without laptop soon, so it might be a good moment making a stop for now. : )
it surely will need some polishing later on.

**Comment by Flohack74** [comment]
Yes so stop here and make a new one if needed. Still waiting on @mariogrip review here ;)

**Pull request merged**